### PR TITLE
Final touches: documentation improvements, minor renames, and import reorgs

### LINKED ISSUE #82: Documentation improvements
Go through the contracts and make sure all functions have up-to-date and comprehensive documentation. Do any updates and reformatting needed.

### PR DESCRIPTION
**Pull request by pdyraga**
Closes #82 
Depends on #124

Some final touches packed in one PR, specifically:
- Solidity documentation improvements: better descriptions, fixed typos, more information on important fields and functions.
- Improvements to Risk Manager interface

  `IRiskManager` has been renamed to `IRiskManagerV1`. Also, added
`notifyLiquidation` and `notifyLiquidated` functions there and improved
docs.

  `IRiskManager` was not a generic interface that could be implemented by
any risk manager because of the notion of signer bonds and the fact they
are getting withdrawn from the risk manager. Instead of having only
functions used by swap strategies there, we can have there all functions
external contracts are interested in.

  Swap strategies still can work with any implementation of a risk manager
for tBTC v1, same as before. They were never generic enough to work with
any risk manager because that manager needed to have a notion of signer
bonds, hence liquidation as well.
- Renamed `RiskManagerV1.GOVERNANCE_TIME_DELAY` to `GOVERNANCE_DELAY `

  Renamed the constant to keep it aligned with "governance delay" used by
the asset pool. Even in the risk manager, we have
onlyAfterGovernanceDelay modifier.

- Reordered and aligned imports 

  First goes internal interfaces. Second goes internal contracts. Then, external interfaces and contracts such as OpenZeppelin ones.

### CODE REVIEW

File: contracts/AssetPool.sol [hunk 1]
```diff
@@ -99,7 +100,7 @@ contract AssetPool is Ownable, IAssetPool {
     /// @notice Reverts if the withdrawl governance delay has not passed yet or
```
Comment (tomaszslabon) [inline contracts/AssetPool.sol:100]: withdrawl -> withdrawal

File: contracts/AssetPool.sol [hunk 2]
```diff
@@ -14,17 +14,17 @@
...
 import "./interfaces/IAssetPoolUpgrade.sol";
 import "./RewardsPool.sol";
 import "./UnderwriterToken.sol";
 import "./GovernanceUtils.sol";
 
-/// @title AssetPool
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
```
Comment (tomaszslabon) [inline contracts/AssetPool.sol:26]: There is a typo in `AssetPool.sol`, but it's not visible here. This PR is probably a good moment to correct it:
`paramter` -> `parameter`

File: contracts/RiskManagerV1.sol [hunk 3]
```diff
@@ -318,7 +381,7 @@ contract RiskManagerV1 is IRiskManager, Auctioneer, Ownable {
     /// @dev Can be called only by the contract owner. The auction length should
     ///      be adjusted very carefully. Total value locked of the coverage pool
     ///      and minimum possible auction amount needs to be taken into account.
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply contracts/RiskManagerV1.sol]: d96abc678f74b75db3de681cf65b6f38b1ee0f11

File: contracts/RiskManagerV1.sol [hunk 4]
```diff
@@ -43,58 +44,103 @@ interface IDeposit {
...
+    //          tBTC deposits that nobody else is willing to purchase bonds
+    ///         from. The value can be updated by the governance in two steps.
+    ///         First step is to begin the update process with the new value
+    ///         and the second step is to finalize it after
+    ///         `GOVERNANCE_DELAY` has passed.
+    uint256 public bondAuctionThreshold; // percentage
     uint256 public newBondAuctionThreshold;
     uint256 public bondAuctionThresholdChangeInitiated;
 
+    /// @notice The length with wich every new auction is opened. Auction length
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply contracts/RiskManagerV1.sol]: d96abc678f74b75db3de681cf65b6f38b1ee0f11

File: contracts/AssetPool.sol [hunk 5]
```diff
@@ -42,21 +42,22 @@ contract AssetPool is Ownable, IAssetPool {
...
-    // After the withdrawal timeout elapses, tokens stay in the pool
-    // and the underwriters has to initiate the withdrawal again and wait for
-    // the full withdrawal delay to complete the withdrawal.
+
+    /// @notice The time the underwriter has after the withdrawal delay passed
+    ///         to complete the withdrawal. During that time, underwriter is
+    ///         still earning rewards and their share of the pool is still
+    ///         a subject of a possible coverage claim.
+    ///         After the withdrawal timeout elapses, tokens stay in the pool
+    ///         and the underwriters has to initiate the withdrawal again and
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply contracts/AssetPool.sol]: d96abc678f74b75db3de681cf65b6f38b1ee0f11

File: contracts/AssetPool.sol [hunk 6]
```diff
@@ -42,21 +42,22 @@ contract AssetPool is Ownable, IAssetPool {
...
     IAssetPoolUpgrade public newAssetPool;
 
-    // The time it takes the underwriter to withdraw their collateral
-    // and rewards from the pool. This is the time that needs to pass between
-    // initiating and completing the withdrawal. During that time, underwriter
-    // is still earning rewards and their share of the pool is still a subject
-    // of a possible coverage claim.
+    /// @notice The time it takes the underwriter to withdraw their collateral
+    ///         and rewards from the pool. This is the time that needs to pass
+    ///         betweenin itiating and completing the withdrawal. During that
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply contracts/AssetPool.sol]: d96abc678f74b75db3de681cf65b6f38b1ee0f11

File: contracts/RewardsPool.sol [hunk 7]
```diff
@@ -14,14 +14,14 @@
...
+
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import "@openzeppelin/contracts/utils/math/Math.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
 
-import "./AssetPool.sol";
-
-/// @title RewardsPool
+/// @title Rewards Pool
```
Comment (tomaszslabon) [inline contracts/RewardsPool.sol:24]: `RewardsPool.sol` contains a typo as well:
immediatelly -> immediately

Comment (pdyraga) [reply contracts/RewardsPool.sol:24]: https://github.com/keep-network/coverage-pools/pull/127/commits/60aa41cfc3d13d044e4bea282377ee435712f981

File: contracts/AssetPool.sol (hunk 2 above)
Comment (pdyraga) [reply contracts/AssetPool.sol:26]: https://github.com/keep-network/coverage-pools/pull/127/commits/60aa41cfc3d13d044e4bea282377ee435712f981

File: contracts/AssetPool.sol (hunk 1 above)
Comment (pdyraga) [reply contracts/AssetPool.sol]: https://github.com/keep-network/coverage-pools/pull/127/commits/60aa41cfc3d13d044e4bea282377ee435712f981

**Review by tomaszslabon — APPROVED** [review]

**Pull request merged**